core/store/src/trie/mod.rs

```diff
@@ -946,9 +946,18 @@ mod tests {
         }
         assert_eq!(pairs, iter_pairs);
 
+        let assert_has_next = |want, other_iter: &mut TrieIterator| {
+            assert_eq!(Some(want), other_iter.next().map(|item| item.unwrap().0).as_deref());
+        };
+
         let mut other_iter = trie.iter().unwrap();
-        other_iter.seek(b"r").unwrap();
-        assert_eq!(other_iter.next().unwrap().unwrap().0, b"x".to_vec());
+        other_iter.seek_prefix(b"r").unwrap();
+        assert_eq!(other_iter.next(), None);
+        other_iter.seek_prefix(b"x").unwrap();
+        assert_has_next(b"x", &mut other_iter);
+        assert_eq!(other_iter.next(), None);
+        other_iter.seek_prefix(b"y").unwrap();
+        assert_has_next(b"y", &mut other_iter);
```

Comment:
Since we’re doing y as well, let’s check None as well.
```suggestion
        assert_has_next(b"y", &mut other_iter);
        assert_eq!(other_iter.next(), None);
```